src/RequestContext.js

```diff
@@ -40,8 +40,15 @@ module.exports = class RequestContext {
     this._wskActivationId = utils.randomChars(32, true);
     this._requestId = utils.randomChars(32);
     this._cdnRequestId = utils.uuid();
-    this._strain = cfg.selectStrain({ ...req, path: this._path });
-    if (this._strain.urls.length > 0) {
+
+    const reqP = { ...req, path: this._path };
+    this._strain = cfg.selectStrain(reqP);
+    if (this._strain.condition) {
```

Comment:
I have a question for you guys: `selectStrain()` finds the matching strain, and returns it. With conditions, the `match()` operation not only returns true/false but also the `baseURL` that matched, which is exactly what the subsequent `mount` calculation requires. Is there a way to "enhance" selectStrain to return the match result as well?